.github/workflows/ci-test-go.yaml

```diff
@@ -0,0 +1,43 @@
+name: ci-test
+
+on:
+  push:
+    branches:
+      - main
+      - stable/*
+      - rc/*
```

Comment:
Do we need this distinction? I understand this is a non-unique change to this PR.

Reply:
I just copied the action and added the sapphire localnet container, given we will likely be changing the actions in another PR to make CI run quicker and the status badges etc more accurately represent the different components I say we address that later